plugins/plugin-chart-echarts/src/Funnel/controlPanel.tsx

```diff
@@ -0,0 +1,205 @@
+              type: 'SelectControl',
+              freeForm: true,
+              label: t('Number format'),
+              renderTrigger: true,
+              default: numberFormat,
+              choices: D3_FORMAT_OPTIONS,
+              description: `${t('D3 format syntax: https://github.com/d3/d3-format')} ${t(
+                'Only applies when "Label Type" is set to show values.',
+              )}`,
+            },
+          },
+        ],
+        [
+          {
+            name: 'show_labels',
+            config: {
+              type: 'CheckboxControl',
+              label: t('Show Labels'),
+              renderTrigger: true,
+              default: showLabels,
+              description: t('Whether to display the labels.'),
+            },
+          },
+        ],
+        [
+          {
+            name: 'label_line',
+            config: {
+              type: 'CheckboxControl',
+              label: t('Label Line'),
+              default: labelLine,
+              renderTrigger: true,
+              description: t('Draw line from Funnel to label when labels outside?'),
+            },
+          },
+          {
+            name: 'min',
+            config: {
+              type: 'SliderControl',
+              label: t('min'),
+              default: min,
+              renderTrigger: true,
+              description: t('The minimum value of the specified data.'),
+            },
+          },
+          {
+            name: 'max',
+            config: {
+              type: 'SliderControl',
+              label: t('min'),
+              default: max,
+              renderTrigger: true,
+              description: t('The maximum value of the specified data.'),
+            },
+          },
+          {
+            name: 'gap',
+            config: {
+              type: 'SliderControl',
+              label: t('gap'),
+              default: gap,
+              renderTrigger: true,
+              description: t('Data graph spacing.'),
+            },
+          },
+          {
+            name: 'sort',
+            config: {
+              type: 'SelectControl',
+              label: t('sort'),
+              default: sort,
+              renderTrigger: true,
+              description: t('Data sorting. The options are ascending, descending, none'),
+            },
+          },
+          {
+            name: 'orient',
+            config: {
+              type: 'SelectControl',
+              label: t('orient'),
+              default: orient,
+              renderTrigger: true,
+              description: t('Funnel chart orientation. The options are vertical, horizontal'),
+            },
+          },
+        ],
+        // eslint-disable-next-line react/jsx-key
+        [<h1 className="section-header">{t('Funnel shape')}</h1>],
+      ],
+    },
+  ],
+  controlOverrides: {
+    series: {
+      validators: [validateNonEmpty],
+      clearable: false,
+    },
+    row_limit: {
+      default: 100,
```

Comment:
Even `100` is probably too many layers. I'd recommend setting it to `20` or `10`.